.pre-commit-config.yaml

```diff
@@ -109,26 +107,42 @@ repos:
   rev: v0.24.1
   hooks:
   - id: validate-pyproject
+    name: validate pyproject.toml
     additional_dependencies: ['validate-pyproject-schema-store[all]']
 
 - repo: https://github.com/tox-dev/pyproject-fmt
   rev: v2.11.0
   hooks:
   - id: pyproject-fmt
+    name: format pyproject.toml
+
+- repo: https://github.com/astral-sh/uv-pre-commit
+  rev: 0.9.4
+  hooks:
+  - id: uv-lock
+    name: update uv.lock if requirements changed
```

Comment:
This hook runs `uv lock`, which will only update `uv.lock` if the requirements in `pyproject.toml` changed. This hook will not __upgrade__ the environment to the most recent versions of dependencies, except where necessary. To do that, run `nox -s requirements`.